modules/apps/transfer/types/transfer_authorization.go

```diff
@@ -166,39 +165,27 @@ func validateMemo(ctx sdk.Context, memo string, allowedPacketDataList []string)
 	}
 
 	// if allowedPacketDataList has only 1 element and it equals AllowAllPacketDataKeys
-	// then accept all the packet data keys
-	if len(allowedPacketDataList) == 1 && allowedPacketDataList[0] == AllowAllPacketDataKeys {
+	// then accept all the memo strings
+	if len(allowedMemos) == 1 && allowedMemos[0] == AllowAllPacketDataKeys {
 		return nil
 	}
 
-	jsonObject := make(map[string]interface{})
-	err := json.Unmarshal([]byte(memo), &jsonObject)
-	if err != nil {
-		return err
-	}
-
 	gasCostPerIteration := ctx.KVGasConfig().IterNextCostFlat
-
-	for _, key := range allowedPacketDataList {
+	isMemoAllowed := slices.ContainsFunc(allowedMemos, func(allowedMemo string) bool {
 		ctx.GasMeter().ConsumeGas(gasCostPerIteration, "transfer authorization")
 
-		_, ok := jsonObject[key]
-		if ok {
-			delete(jsonObject, key)
+		if strings.TrimSpace(memo) == strings.TrimSpace(allowedMemo) {
```

Comment:
Thinking more about this, I think trim spaces doesn't make much sense in here. We are allowing a specific memo string, and not making any assumptions about the format such as it being json. It makes no sense that if a user allowed `"rent payment"`, then they also allow `"rentpayment"`. Especially if these transactions would be indexed by an indexer later based on the memo.

Maybe worth asking others, but I'm starting to feel more strongly about this.

Reply:
`TrimSpace` only trims leading and trailing spaces, not spaces in between the text. That's why I thought that it would make sense to use it.

Reply:
Ah ok, I'm back to being indifferent about it then. Still ran be removed imo. We shouldn't be cleaning up user mess XD